### ISSUE TITLE
Using deep ... -1 with nested data does not return unliminted nested objects

### ISSUE THREAD

**Issue by andriusign** [directus/directus#6900]
### Describe the Bug

Getting articles from **/items/articles** returns data with nested (4 comments as expected):
```
  "comments": [
            6,
            10,
            19,
            20
        ]
```
Getting articles from **/items/articles?deep={"comments":{"_limit":1}}** returns data with nested (1 comment as expected):
```
 "comments": [
            6
        ]
```
Getting articles from **/items/articles?deep={"comments":{"_limit":-1}}** returns data with nested (3 comments returned, 4 expected):
```
"comments": [
            6,
            10,
            19
        ]
```

Trying to get more than 100 nested items according to example https://docs.directus.io/reference/api/query/#examples-7 but -1 seems not to work (returns unexpected number of items). By default only getting 100 nested object items.

### To Reproduce

Call /items/articles?deep={"comments":{"_limit":-1}} with **deep** parameter.

### What version of Directus are you using?

v9.0.0-rc.70

### What version of Node.js are you using?

v14.15.3

### What database are you using?

Postgresql

### What browser are you using?

Any

### What operating system are you using?

Linux

### How are you deploying Directus?

Docker

**Comment by paescuj** [comment]
Fixed in [v9.0.0-rc.73](https://github.com/directus/directus/blob/main/changelog.md#v900-rc73-june-4-2021) by #6054.
See https://docs.directus.io/guides/projects/#upgrading-updating-a-project for upgrade guide.

**Comment by andriusign** [comment]
Thanks @paescuj , seems to have solved the issue after upgrading to rc85.
